mmv1/third_party/terraform/tests/resource_datapipeline_pipeline_test.go

```diff
@@ -0,0 +1,150 @@
+	rs, ok := s.RootModule().Resources[res]
+	if !ok {
+		return "", fmt.Errorf("not found: %s", res)
+	}
+
+	if rs.Primary.ID == "" {
+		return "", fmt.Errorf("no ID is set for %s", res)
+	}
+
+	if v, ok := rs.Primary.Attributes["id"]; ok {
+		return v, nil
+	}
+
+	return "", fmt.Errorf("id not set on resource %s", res)
+}
+
+func TestAccDataPipelinePipeline_basic(t *testing.T) {
+	t.Parallel()
+
+	var generatedId string
+	VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckDataPipelinePipelineDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataPipelinePipeline_basic(),
+				Check:  setTestCheckDataPipelinePipelineId("google_data_pipeline_pipeline.primary", &generatedId),
+			},
+			{
+				ResourceName:      "google_data_pipeline_pipeline.primary",
+				ImportState:       true,
+				ImportStateVerify: true,
+				// Ignore input-only field for import
+				ImportStateVerifyIgnore: []string{"region"},
+			},
+			{
+				Config: testAccDataPipelinePipeline_basicUpdate(),
+				Check:  testCheckDataPipelinePipelineIdAfterUpdate("google_data_pipeline_pipeline.primary", &generatedId),
+			},
+			{
+				ResourceName:      "google_data_pipeline_pipeline.primary",
+				ImportState:       true,
+				ImportStateVerify: true,
+				// Ignore input-only field for import
+				ImportStateVerifyIgnore: []string{"region"},
+			},
+		},
+	})
+}
+
+func testAccDataPipelinePipeline_basicUpdate() string {
+	return `
+resource "google_data_pipeline_pipeline" "primary" {
+  name         = "my-pipeline"
+  display_name = "update-pipeline"
+  type         = "PIPELINE_TYPE_BATCH"
+  state        = "STATE_ACTIVE"
+
+  workload {
+    dataflow_flex_template_request {
+      project_id = "my-project"
+      launch_parameter {
+        job_name = "my-job"
+        environment {
+          temp_location = "gs://my-bucket/tmp_dir"
+        }
+        container_spec_gcs_path = "gs://my-bucket/templates/template_file"
+      }
+      location      = "us-central1"
+      validate_only = true
+    }
+  }
+  schedule_info {
+    schedule  = "* */2 * * *"
+    time_zone = "UTC"
+  }
+}
+`
+}
+func testAccDataPipelinePipeline_basic() string {
+	return `
+resource "google_data_pipeline_pipeline" "primary" {
+  name         = "my-pipeline"
+  type         = "PIPELINE_TYPE_BATCH"
+  state        = "STATE_ACTIVE"
+
+  workload {
+    dataflow_flex_template_request {
+      project_id = "my-project"
+      launch_parameter {
+        job_name = "my-job"
+        environment {
+          temp_location = "gs://my-bucket/tmp_dir"
+        }
+        container_spec_gcs_path = "gs://my-bucket/templates/template_file"
+      }
+      location      = "us-central1"
+      validate_only = true
+    }
+  }
+  schedule_info {
+    schedule  = "* */2 * * *"
+    time_zone = "UTC"
+  }
+}
+`
+}
```

Comment:
Can you modify the resource name to have tf-test prepended 

See our other tests
https://github.com/GoogleCloudPlatform/magic-modules/blob/2e90d01aef7f2973ec76c98926aba055cd45e64e/mmv1/third_party/terraform/tests/resource_compute_router_interface_test.go#L53C3-L74